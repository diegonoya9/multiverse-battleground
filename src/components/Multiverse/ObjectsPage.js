import classes from "./ObjectsPage.module.css"
import { memo, useEffect, useState } from "react";
import ReactAudioPlayer from 'react-audio-player';
import musicFile from "../../assets/sounds/music/DiscoEterno.WAV"
const ObjectsPage = ({ user, changeMultiverseActivePage }) => {
    const audioStyle = {
        display: 'none',
    };
    return (<div>
        <ReactAudioPlayer src={musicFile} /*autoPlay*/ controls style={audioStyle} />
        <button value="Back to Main Menu" className={classes.backToMainMenuBtn} onClick={() => { changeMultiverseActivePage("mainMenu") }} >Back to Main Menu </button>
        <div className={classes.container} >
            {user &&
                user.objects.map((object) => {
                    return (
                        <div className={classes.objectContainer} key={object.name}>
                            <span className={classes.objectName}>{object.name}</span>
                            <div className={classes.imageContainer}>
                                <img alt="fighter" src={object.img} className={classes.objectImg} />
                            </div>
                            <span className={classes.objectName}>Quantity:{object.quantity}</span>
                        </div>
                    );
                })}
        </div>
    </div>
    );

}

export default memo(ObjectsPage)