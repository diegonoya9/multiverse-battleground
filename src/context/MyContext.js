import React, { createContext, useState } from 'react';

const MyContext = createContext();

const MyContextProvider = ({ children }) => {
    const [userContext, setUserContext] = useState({
        idUsuario: 1,
        userName: "",
        backEndUrl: process.env.NODE_ENV === 'development' ? "http://localhost:3009/api" : "https://multiverse-battlegorund-back.onrender.com/api" // Otros datos relacionados con el usuario si es necesario
    });

    const setUserId = (newId) => {
        setUserContext({
            ...userContext,
            idUsuario: newId,
        });
    };
    const setUserName = (userName) => {
        setUserContext({
            ...userContext,
            userName
        });
    };

    return (
        <MyContext.Provider value={{ userContext, setUserId, setUserName }}>
            {children}
        </MyContext.Provider>
    );
};

export { MyContext, MyContextProvider };
