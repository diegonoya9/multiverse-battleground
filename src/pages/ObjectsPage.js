import classes from "./ObjectsPage.module.css"
import { memo, useEffect, useState, useContext } from "react";
import Button from "../components/UI/Button";
import ObjectCard from "../components/UI/ObjectCard";
import { MyContext } from "../context/MyContext";
import { useNavigate } from "react-router-dom";
const ObjectsPage = () => {
    const navigate = useNavigate()
    const [userObjects, setUserObjects] = useState()
    const { userContext } = useContext(MyContext);
    let backEndUrl = userContext.backEndUrl
    let user = userContext.user
    useEffect(() => {
        const fetchData = async () => {
            try {
                if (backEndUrl && user) {
                    const response = await fetch(backEndUrl + '/alluserobjects/' + user.user_id);
                    const data = await response.json();
                    setUserObjects(data);
                }
            } catch (error) {
                console.error('Error fetching data:', error);
            }
        };

        fetchData();
    }, [backEndUrl, user])
    return (<div className={classes.backgroundImg}>
        <Button colorType="lightgreen" value="Back to Main Menu" onClick={() => { navigate('/') }}></Button>
        <div className={classes.container} >
            {userObjects &&
                userObjects.map((object) => {
                    return (
                        object.objects.name !== "Money" &&
                        <div className={classes.objectContainer} key={object.name}>
                            <ObjectCard showPrice={false} object={object}></ObjectCard>
                        </div>
                    );
                })}
        </div>
    </div>
    );

}

export default memo(ObjectsPage)