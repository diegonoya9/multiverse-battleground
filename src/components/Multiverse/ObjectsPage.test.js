// ObjectsPage.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MyContextProvider } from '../../context/MyContext';
import ObjectsPage from './ObjectsPage';
afterEach(() => {
  global.fetch.mockRestore();
  cleanup();
});
// Mock del contexto para la prueba
const mockUserContext = {
  idUsuario: 1,
  backEndUrl: "http://localhost:3009/api"
  // Otros datos relacionados con el usuario si es necesario
};
const objects = [
  {
    "user_object_id": 2,
    "user_id": 1,
    "object_id": 7,
    "quantity": 25911,
    "objects": {
      "object_id": 7,
      "category": "userItem",
      "description": "Money",
      "img": "./assets/img/money.png",
      "name": "Money",
      "price": 1,
      "quantity": 1,
      "type": "consumable",
      "actionobjects": []
    },
    "name": "Money",
    "description": "Money",
    "img": "./assets/img/money.png"
  }
]
const user = { user_id: 1 }
test('renders objects page component', () => {
  global.fetch = jest.fn().mockResolvedValue({
    json: jest.fn().mockResolvedValue(objects)
  });
  // Renderiza ObjectsPage dentro de MyContextProvider con el contexto simulado
  const { getByText } = render(
    <MyContextProvider value={{ userContext: mockUserContext }}>
      <ObjectsPage user={user} />
    </MyContextProvider>
  );

  // Puedes agregar expectativas para asegurarte de que los elementos esperados estï¿½n presentes
  expect(getByText(/Back to Main Menu/)).toBeInTheDocument();
});
