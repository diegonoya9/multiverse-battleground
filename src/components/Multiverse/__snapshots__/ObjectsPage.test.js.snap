// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders objectpage correctly according to snapshot 1`] = `
<div
  className="backgroundImg"
>
  <audio
    autoPlay={true}
    className="react-audio-player "
    controls={true}
    id=""
    loop={false}
    muted={false}
    preload="metadata"
    src="DiscoEterno.WAV"
    style={
      Object {
        "display": "none",
      }
    }
    title="DiscoEterno.WAV"
  >
    <p>
      Your browser does not support the 
      <code>
        audio
      </code>
       element.
    </p>
  </audio>
  <div
    className="button undefined undefined"
    onClick={[Function]}
  >
    Back to Main Menu
  </div>
  <div
    className="container"
  />
</div>
`;
